modules/core/src/main/scala/com/caesars/tracing/ZTracer.scala

```diff
@@ -0,0 +1,85 @@
+package com.caesars.tracing
+
+import cats.data.NonEmptyList
+import io.janstenpickle.trace4cats.inject.EntryPoint
+import io.janstenpickle.trace4cats.model.*
+import io.janstenpickle.trace4cats.{ErrorHandler, Span}
+import zio.*
+import zio.interop.catz.*
+
+//TODO: find away to remove the option
+//We have the following problem:
+//In application code, we should be blissfully unaware of what spans have been created before and where we are in the structure
+//Specifically, you should not need to know whether you are creating a root span or a child span
+//Also, applications may have more than one root span
+
+//What we would like, is a ZTracer that always has a span, and can perform `context`, `setStatus`, `putAll` and `span` safely
+final case class ZTracer private (
+    private val currentSpan: FiberRef[Option[ZSpan]],
```

Comment:
Could you please explain the choice of `FiberRef`? I'm asking since I don't know what it is. I will read up on it, but curious of why this structure is used, say, over `Ref`. Thanks

Reply:
Hey Kevin - FiberRefs track state across each fiber where each fiber has access to its own Ref. Here we use this feature to keep track of sub-spans (that can occur concurrently) and make sure that the child picks up and continues the span from its parent.